7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -26,34 +10,170 @@ split_lake_centroids <- function(centroids_sf_rds) {
     sample_n(5)
 }
 
+# Reproject to the specified crs
+project_to_crs <- function(sf_obj, crs) {
```

Comment:
I'm not sure this function is needed. The default function signature for `st_transform` is 
```r
st_transform(
  x,
  crs = st_crs(x),
  ...,
```
so it is already doing what your function does. Can you replace `sf::st_transform` where you call `project_to_crs` and produce identical results?

Reply:
Yes, but Lindsay asked me to clean up my `_targets.R` file, so I was trying to shorten the code in there by writing this function. Will revert.

Reply:
I think cleaning up in this case would mean that this re-projection could be grouped into an existing function, rather than a standalone target. It might not make sense to make the reprojected sf a standalone sf object but instead do the reprojection as part of a function that preps the sf object. I was feeling like the current code was almost _too_ modularized, where each step was a different target. Since `targets` checks the status of _every_ target when it rebuilds, this can cause some unnecessary lagginess, so I am advocating for grouping more of the targets into logical chunks.

Reply:
I think we can do that as a step later, too. We could add an issue for grouping the targets and circle back to it. I'd be happy to take the reigns on that effort.

Reply:
Agreed. Keep in mind that `targets` (or `remake` would be doing this too) is saving a copy of the target behind the scenes for each one. So if you've got one target for the raw spatial, another for the re-projected, and another for a filtered version, there are lots of near-copies being stored locally. One thing to keep in mind with how finely you split things up.

Reply:
Yes, @lindsayplatt made that same point, Jordan, and I think I do need to think through it more and make some additional changes. I was thinking I had to have a separate file target initially, in order for `targets` to track changes to that file (since it's not created in the repo), but maybe I'm mistaken.

Reply:
You are doing the file tracking correctly - a single target that is only the filepath with `format = "file"` and then subsequent targets to read and re-project. I think in this case, rather than reprojecting the lake centroids to the grid CRS as its own target, you could pass in `subset_lake_centroids_sf` to `get_tile_cells()` and reproject within that function. Again, I am happy to table this and think of the grouping as a separate issue to tackle, so that we can keep moving forward.

Reply:
K I'm making some changes now - I was wrong about `sf`, seems to work fine?